### PR TITLE
Add CLI option for custom config file paths

### PR DESCRIPTION
**Pull request by milkey-mouse**
Added a section in the docs for it too. It overrides the generic-named config files in the custom file's folder.

### CODE REVIEW

File: src/pydocstyle/config.py [hunk 1]
```diff
@@ -195,7 +195,20 @@ def _get_config(self, node):
...
+            config_file = self._get_config_file_in_folder(path)
+        else:
+            # Look for the config file from the command line option.
+            cli_config_dir = os.path.dirname(self._options.config_location)
+            if os.path.abspath(cli_config_dir) == os.getcwd():
+                # We're in the specified config file's directory.
+                # Take that one instead of searching for a generic name.
+                config_file = self._options.config_location
+            else:
+                # Search anyway since we're not in the right directory.
```
Comment (Nurdok) [inline src/pydocstyle/config.py]: I don't understand this logic. Why do you only allow configuration files from the current directory?

**Comment by Nurdok** [comment]
@milkey-mouse are you still interested in working on this PR?

**Comment by milkey-mouse** [comment]
@Nurdok I lost access to my GitHub account for the last few days. I'll work on it.

**Comment by jacebrowning** [comment]
@milkey-mouse Still working on this? If not, I'd like to take a stab at it.

**Comment by milkey-mouse** [comment]
Ok, go ahead. I got off track anyway.

**Comment by elaichi** [comment]
@jacebrowning @milkey-mouse bump :)

**Comment by jacebrowning** [comment]
@elaichi I'm not working on this anymore. Feel free to pick up where I left off in https://github.com/PyCQA/pydocstyle/pull/236. It might be worth discussing implementation before working because there was some disagreement on how this feature should function.

**Comment by Nurdok** [comment]
I'm closing this PR as part of cleaning up stale issues and PRs. Feel free to open if you believe this is still relevant. Thanks!

**Pull request closed without merging**